### PR TITLE
fix: run context handling for aws-sdk DynamoDB instrumentation

### PR DESCRIPTION
**Pull request by trentm**
This also adds a couple aws-sdk-related tests to the TAV=aws-sdk tests.

Refs: #2430

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/master/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

**Comment by trentm** [comment]
@elasticmachine run elasticsearch-ci/docs skiplinkcheck

File: .tav.yml [hunk 1]
```diff
@@ -479,6 +479,8 @@ aws-sdk:
   # Test v2.858.0, every N=31 of 159 releases, and current latest.
   versions: '2.858.0 || 2.889.0 || 2.920.0 || 2.951.0 || 2.982.0 || 2.1013.0 || 2.1016.0 || >2.1016.0 <3'
   commands:
+    - node test/instrumentation/modules/aws-sdk/aws4-retries.test.js
```
Comment (astorm) [inline .tav.yml:482]: +1 for getting these tests into the tav runner while we're in here.

File: test/instrumentation/modules/aws-sdk/dynamodb.test.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+'use strict'
```
Comment (astorm) [inline test/instrumentation/modules/aws-sdk/dynamodb.test.js:1]: +1 here -- it'd be nice if our linter/CI caught these though (not for this PR, just something to keep in mind for the future)

File: .tav.yml (hunk 1 above)
Comment (astorm) [inline .tav.yml:482]: +1 for getting these tests into the tav runner while we're in here.

**Review by astorm — APPROVED** [review]
+1 approved -- looks like a pretty now-standard context swap.

File: test/instrumentation/modules/aws-sdk/dynamodb.test.js (hunk 2 above)
Comment (trentm) [reply test/instrumentation/modules/aws-sdk/dynamodb.test.js:1]: Yah, agreed. I *think* we can further configure our `standard` usage with eslint rules?  https://eslint.org/docs/rules/strict

**Pull request merged**